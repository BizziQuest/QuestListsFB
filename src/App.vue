<template>
  <v-app>
    <p> {{message}} </p>
    <v-btn class="primary" v-on:click="messageIs()"> what is message? </v-btn>
    <v-alert
      :value = "messageValue"
      type = "success"
    >
      This is a success alert.
    </v-alert>
  </v-app>
</template>


<script>

export default {
  components: {},
  data() {
    return {
      messageValue: false,
      message: 'Vuetify is here',
    };
  },
  methods: {
    messageIs() {
      this.messageValue = !this.messageValue;
    },
  },
};
</script>

<style scoped>
p {
  font-size: 2em;
  text-align: center;
}
</style>
