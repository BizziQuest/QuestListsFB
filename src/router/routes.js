import EditInfo from '../components/EditInfo.vue';
import Lists from '../views/Lists.vue';
import List from '../views/List.vue';

const routes = [
  {
    path: '/',
    name: 'Lists',
    component: Lists,
  },
  {
    path: '/Lists/:slug+',
    name: 'List',
    component: List,
  },
  {
    path: '/about',
    name: 'About',
    component: () => import(/* webpackChunkName: "about" */ '../views/About.vue'),
  },
  // {
  //   path: '/about/terms',
  //   name: 'Terms',
  //   component: () => import(/* webpackChunkName: "tos" */ '../views/about/Terms.vue'),
  // },
  // {
  //   path: '/about/conduct',
  //   name: 'Conduct',
  //   component: () => import(/* webpackChunkName: "conduct" */ '../views/about/Conduct.vue'),
  // },
  // {
  //   path: '/about/privacy',
  //   name: 'Privacy',
  //   component: () => import(/* webpackChunkName: "privacy" */ '../views/about/Privacy.vue'),
  // },
  {
    path: '/editinfo',
    name: 'EditInfo',
    component: EditInfo,
  },
];

export default routes;
