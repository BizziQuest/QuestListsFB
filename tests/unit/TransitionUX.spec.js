import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Vuetify from 'vuetify';
import App from '@/App.vue';
import VueRouter from 'vue-router';
import routes from '@/router/routes';

const localVue = createLocalVue();
const router = new VueRouter({ routes });
const vuetify = new Vuetify();
localVue.use(VueRouter);
localVue.use(Vuex);
localVue.use(Vuetify);

let wrapper;

beforeEach(() => {
  const state = {
    messages: [],
    currentUser: {
      avatar: '/img/unknown_user.svg', // when we set the user in the store, we default to this.
      displayName: null,
      email: 'tersterson3@test.com',
      emailVerified: false,
      uid: 'UUID123456',
      useGravatar: false,
    },
    globalPreferences: {},
  };
  const localStore = new Vuex.Store({
    getters: { getGlobalPreferences: (s) => s.globalPreferences },
    state,
    actions: { fetchLists: jest.fn() },
  });
  wrapper = mount(App, {
    localVue,
    router,
    vuetify,
    store: localStore,
  });
});

afterEach(() => {
  wrapper.destroy();
});

describe('smoother ux transition', () => {
  it('has a transition component', () => {
    const wrapper1 = shallowMount(App, {
      localVue,
      router,
      vuetify,
    });
    expect(wrapper1.find('transition-stub[name="router-anim"]').exists()).toBe(true);
  });

  describe('transitioning between lists view and edit info page (should be less than 0.3 seconds)', () => {
    it.todo('(find out how to test transitions) lists view should have the proper classes when leaving');
    it.todo('(find out how to test transitions) profile view should have the proper classes when entering');
  });
});
