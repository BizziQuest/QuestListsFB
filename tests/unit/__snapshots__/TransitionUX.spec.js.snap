// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoother ux transition transitioning between lists view and edit info page (should be less than 0.3 seconds) renders correctly 1`] = `
<div
  class="v-application v-application--is-ltr theme--light"
  data-app="true"
  id="app"
>
  <div
    class="v-application--wrap"
  >
    <nav
      class="v-navigation-drawer v-navigation-drawer--fixed v-navigation-drawer--mini-variant v-navigation-drawer--open v-navigation-drawer--open-on-hover theme--dark"
      style="height: 100vh; top: 0px; max-height: calc(100% - 0px); transform: translateX(0%); width: 56px;"
      test-navigation-drawer=""
    >
      <div
        class="v-navigation-drawer__content"
      >
        <div
          class="v-list v-sheet theme--dark v-list--dense v-list--nav"
        >
          <div
            class="user-menu-item"
          >
            <a
              class="v-list-item v-list-item--link theme--dark"
              href="#/EditInfo"
              tabindex="0"
              title="User Management"
            >
              <div
                class="v-avatar v-list-item__avatar"
                style="height: 28px; min-width: 28px; width: 28px;"
              >
                <div
                  class="v-image v-responsive theme--dark"
                >
                  <transition-stub
                    mode="in-out"
                    name="fade-transition"
                  >
                    <div
                      class="v-image__image v-image__image--preload v-image__image--cover"
                      style="background-position: center center;"
                    />
                  </transition-stub>
                  <div
                    class="v-responsive__content"
                  />
                </div>
              </div>
               
              <div
                class="v-list-item__content ml-3"
              >
                <div
                  class="v-list-item__title text-truncate"
                >
                  
                </div>
              </div>
               
              <button
                class="v-icon notranslate v-icon--link mdi mdi-exit-to-app theme--dark"
                icon=""
                style="font-size: 16px;"
                type="button"
              />
            </a>
          </div>
           
          <a
            aria-current="page"
            class="v-list-item--active v-list-item v-list-item--link theme--dark secondary--text"
            href="#/"
            tabindex="0"
            test-questlists-link=""
            title="View All QuestLists"
          >
            <div
              class="v-list-item__action"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-shield-check-outline theme--dark"
              />
            </div>
             
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title"
              >
                QuestLists
              </div>
            </div>
          </a>
           
          <div
            class="v-dialog__container primary"
            role="dialog"
            test-create-list=""
          >
            <!---->
          </div>
          <div
            class="v-list-item v-list-item--link theme--dark secondary--text"
            tabindex="0"
            test-default-create-list-item=""
            title="Create A New Quest"
          >
            <div
              class="v-list-item__action"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate material-icons theme--dark secondary--text"
              >
                add
              </i>
            </div>
             
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title secondary--text"
                test-activate-item=""
              >
                New Quest
              </div>
            </div>
          </div>
           
          <a
            class="v-list-item v-list-item--link theme--dark"
            href="#/search"
            tabindex="0"
            test-search-link=""
            title="Look for a particular quest"
          >
            <div
              class="v-list-item__action"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-magnify theme--dark"
              />
            </div>
             
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title"
              >
                Search
              </div>
            </div>
          </a>
           
          <div
            class="v-list-item theme--dark"
            tabindex="-1"
            test-fav-header=""
          >
            <hr
              aria-orientation="horizontal"
              class="my-4 v-divider theme--dark"
              role="separator"
            />
             
            <div
              class="v-subheader theme--dark"
            >
              Favorite Quests
            </div>
          </div>
           
          <a
            class="v-list-item v-list-item--link theme--dark"
            href="#/lists/0"
            tabindex="0"
            test-fav-link=""
            title="Sample Favorite Quest with amazingly long title and now that's about it"
          >
            <div
              class="v-list-item__action"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-heart theme--dark"
              />
            </div>
             
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title"
              >
                Sample Favorite Quest with amazingly long title and now that's about it
              </div>
            </div>
          </a>
           
          <div
            class="v-list-item theme--dark"
            tabindex="-1"
          >
            <hr
              aria-orientation="horizontal"
              class="my-4 v-divider theme--dark"
              role="separator"
            />
             
            <div
              class="v-subheader theme--dark"
            />
          </div>
           
          <div
            class="v-list-item v-list-item--link theme--dark"
            tabindex="0"
            test-dark-mode-switch=""
          >
            <div
              class="v-list-item__action"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-brightness-7 theme--dark"
              />
            </div>
             
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title"
              >
                Turn Dark Mode On
              </div>
            </div>
          </div>
           
          <a
            class="v-list-item v-list-item--link theme--dark"
            href="#/about"
            tabindex="0"
            title="About QuestLists"
          >
            <div
              class="v-list-item__action"
            >
              <i
                aria-hidden="true"
                class="v-icon notranslate mdi mdi-information theme--dark"
              />
            </div>
             
            <div
              class="v-list-item__content"
            >
              <div
                class="v-list-item__title"
              >
                About QuestLists
              </div>
            </div>
          </a>
        </div>
      </div>
      <div
        class="v-navigation-drawer__border"
      />
    </nav>
     
    <main
      class="v-main"
      style="padding: 0px 0px 0px 56px;"
    >
      <div
        class="v-main__wrap"
      >
        <transition-stub
          name="router-anim"
        >
          <div
            class="container lists-view container--fluid"
          >
            <transition-group-stub
              class="row"
              name="fade"
              tag="div"
            >
              <!---->
               
            </transition-group-stub>
          </div>
        </transition-stub>
      </div>
    </main>
     
    <!---->
     
    <div>
      <!---->
    </div>
  </div>
</div>
`;
